File: BuilderDesignPattern/Desktop_Example/DesktopBuilder.java
package BuilderDesignPattern.Desktop_Example;

abstract class DesktopBuilder {
	protected Desktop desktop;

	public DesktopBuilder() {
		this.desktop = new Desktop();
	}

	public abstract void buildMonitor();

	public abstract void buildKeyboard();

	public abstract void buildMouse();

	public abstract void buildSpeakers();

	public abstract void buildRam();

	public abstract void buildProcessor();

	public abstract void buildMotherboard();

	public Desktop getDesktop() {
		return this.desktop;
	}

}

File: BuilderDesignPattern/Destop_Example_Python/Builders/DesktopBuilder.py
from abc import ABC, abstractmethod

from BuilderDesignPattern.Destop_Example_Python.Products.Desktop import Desktop


class DesktopBuilder(ABC):
	_desktop: Desktop

	def __init__(self):
		self._desktop = Desktop()

	@abstractmethod
	def buildMonitor(self):
		pass

	@abstractmethod
	def buildKeyboard(self):
		pass

	@abstractmethod
	def buildMouse(self):
		pass

	@abstractmethod
	def buildSpeakers(self):
		pass

	@abstractmethod
	def buildRam(self):
		pass

	@abstractmethod
	def buildProcessor(self):
		pass

	@abstractmethod
	def buildMotherboard(self):
		pass

	def getDesktop(self) -> Desktop:
		return self._desktop
